app/gui2/src/util/__tests__/array.test.ts

```diff
@@ -14,18 +14,20 @@ fcTest.prop({
 })
 
 fcTest.prop({
-  arr: fc
-    .array(fc.float())
-    .map((a) => ({ arr: a.sort((a, b) => a - b), i: Math.floor(Math.random() * a.length) })),
+  arr: fc.array(fc.float({ noNaN: true })).map((a) => ({
+    arr: a.sort((a, b) => a - b),
+    i: Math.max(0, a.indexOf(a[Math.floor(Math.random() * a.length)]!)),
+  })),
 })('partitionPoint (ascending)', ({ arr: { arr, i } }) => {
   const target = arr[i]!
   expect(partitionPoint(arr, (n) => n < target)).toEqual(i)
 })
 
 fcTest.prop({
-  arr: fc
-    .array(fc.float())
-    .map((a) => ({ arr: a.sort((a, b) => b - a), i: Math.floor(Math.random() * a.length) })),
+  arr: fc.array(fc.float({ noNaN: true })).map((a) => ({
+    arr: a.sort((a, b) => b - a),
+    i: Math.max(0, a.indexOf(a[Math.floor(Math.random() * a.length)]!)),
```

Comment:
I've missed it in previous review, but it is still important and may cause this test to fail without proper error report. The usage of `random` is still incorrect here. There should never be any `Math.random` in a test, even in properties functions. That prevents the prop-test from properly shrinking the test cases (since test fail/success is dependant on a factor outside of its control).
To get around that, you could do one of following thing:
- use either a [chain or map](https://fast-check.dev/docs/core-blocks/arbitraries/combiners/any/#chain) to generate the `i` property without using `Math.random`
- Generate an `f` float arbitrary with a known range (e.g. `[0, 1)`), then map it into the appropriate array index in the test body itself
```ts
const i = Math.floor(f * arr.length)
```
That way the index is always valid and deterministic, and still totally within the control of fast-check, so it can be properly reduced.